test/openai/test_completion.py

```diff
@@ -133,14 +133,381 @@ def success_metrics(responses, prompt, test, entry_point):
+    test_data = [
+        {
+            "problem": test_data[x]["problem"],
+            "solution": test_data[x]["solution"],
+        }
+        for x in range(len(test_data))
+        if test_data[x]["level"] == "Level 1"
+    ]
+    print(
+        "max tokens in tuning data's canonical solutions",
+        max([len(x["solution"].split()) for x in tune_data]),
+    )
+    print(len(tune_data), len(test_data))
+    # prompt template
+    prompts = [
+        lambda data: "Given a mathematics problem, determine the answer. Simplify your answer as much as possible.\n###\nProblem: What is the value of $\\sqrt{3! \\cdot 3!}$ expressed as a positive integer?\nAnswer: $\\sqrt{3!\\cdot3!}$ is equal to $\\sqrt{(3!)^2}=3!=3\\cdot2\\cdot1=\\boxed{6}$.\n###\nProblem: %s\nAnswer:"
+        + data["problem"]
+    ]
+
+    try:
+        oai.ChatCompletion.set_cache(seed)
+        vanilla_config = {
+            "model": "gpt-3.5-turbo",
+            "temperature": 1,
+            "max_tokens": 2048,
+            "n": 1,
+            "prompt": prompts[0],
+            "stop": "###",
+        }
+        test_data_sample = test_data[0:3]
+        result = oai.ChatCompletion.test(
+            test_data_sample, vanilla_config, success_metrics
+        )
+        test_data_sample = test_data[3:6]
+        result = oai.ChatCompletion.test(
+            test_data_sample,
+            vanilla_config,
+            success_metrics,
+            use_cache=False,
+            agg_method="median",
+        )
+
+        def my_median(results):
+            return np.median(results)
+
+        def my_average(results):
+            return np.mean(results)
+
+        result = oai.ChatCompletion.test(
+            test_data_sample,
+            vanilla_config,
+            success_metrics,
+            use_cache=False,
+            agg_method=my_median,
+        )
+        result = oai.ChatCompletion.test(
+            test_data_sample,
+            vanilla_config,
+            success_metrics,
+            use_cache=False,
+            agg_method={"expected_success": my_median, "success": my_average},
+        )
+
+        print(result)
+
+        config, _ = oai.ChatCompletion.tune(
+            data=tune_data,  # the data for tuning
+            metric="expected_success",  # the metric to optimize
+            mode="max",  # the optimization mode
+            eval_func=success_metrics,  # the evaluation function to return the success metrics
+            # log_file_name="logs/math.log",  # the log file name
+            inference_budget=0.002,  # the inference budget (dollar)
+            optimization_budget=0.01,  # the optimization budget (dollar)
+            num_samples=num_samples,
+            prompt=prompts,  # the prompt templates to choose from
+            stop="###",  # the stop sequence
+        )
+        print("tuned config", config)
+        result = oai.ChatCompletion.test(test_data_sample, config)
+        print("result from tuned config:", result)
```

Comment:
Are we going to compare the tuned result and the untuned result?

Reply:
We are going to compare in the chatgpt notebook. In this test file, just testing whether we can evaluate the performance of a config without tuning and the performance of the tuned config.

Reply:
Sure. Also, this test file is very long. It seems a lot of code were copied and pasted from https://github.com/hendrycks/math. Is it possible to make the test file short (like the other test files)? For example using a utils function or directly importing from hendrycks.

Reply:
> Sure. Also, this test file is very long. It seems a lot of code were copied and pasted from https://github.com/hendrycks/math. Is it possible to make the test file short (like the other test files)? For example using a utils function or directly importing from hendrycks.

did that in #968